File: create_structure.py
import os
import sys

base_dir = r'd:\Projects\cigar_box_production'

# Change to base directory
os.chdir(base_dir)

dirs = [
    'src/app',
    'src/components',
    'src/components/order',
    'src/components/planning',
    'src/types',
    'src/utils',
    'src/api',
    'src/api/routes',
    'public'
]

files = {
    'src/app/index.ts': '// Placeholder file',
    'src/components/index.ts': '',
    'src/components/order/index.ts': '',
    'src/components/planning/index.ts': '',
    'src/types/index.ts': '',
    'src/utils/index.ts': '',
    'src/api/index.ts': '',
    'src/api/routes/index.ts': '',
    'public/.gitkeep': ''
}

# Create directories
print('Creating directories...')
for d in dirs:
    os.makedirs(d, exist_ok=True)
    print(f'Created: {d}')

# Create files
print('\nCreating files...')
for file_path, content in files.items():
    with open(file_path, 'w') as f:
        f.write(content)
    print(f'Created: {file_path}')

print('\n========== SUMMARY ==========')
print('Directories created: 9')
print('TypeScript files created: 8')
print('Additional files: 1 (.gitkeep)')
print('Total items: 18')
print('\n✓ All directories and files created successfully!')

# List directory structure
print('\n========== DIRECTORY STRUCTURE ==========')
for root, dirs_list, files_list in os.walk('.'):
    level = root.replace('.', '').count(os.sep)
    indent = ' ' * 2 * level
    print(f'{indent}{os.path.basename(root)}/')
    subindent = ' ' * 2 * (level + 1)
    for file in files_list:
        print(f'{subindent}{file}')

File: create_structure.js
const fs = require('fs');
const path = require('path');

const baseDir = 'd:\\Projects\\cigar_box_production';

const dirs = [
  'src/app',
  'src/components',
  'src/components/order',
  'src/components/planning',
  'src/types',
  'src/utils',
  'src/api',
  'src/api/routes',
  'public'
];

const files = {
  'src/app/index.ts': '// Placeholder file',
  'src/components/index.ts': '',
  'src/components/order/index.ts': '',
  'src/components/planning/index.ts': '',
  'src/types/index.ts': '',
  'src/utils/index.ts': '',
  'src/api/index.ts': '',
  'src/api/routes/index.ts': '',
  'public/.gitkeep': ''
};

// Create directories
console.log('Creating directories...');
dirs.forEach(d => {
  const fullPath = path.join(baseDir, d);
  fs.mkdirSync(fullPath, { recursive: true });
  console.log(`Created: ${d}`);
});

// Create files
console.log('\nCreating files...');
Object.entries(files).forEach(([filePath, content]) => {
  const fullPath = path.join(baseDir, filePath);
  fs.writeFileSync(fullPath, content);
  console.log(`Created: ${filePath}`);
});

console.log('\n========== SUMMARY ==========');
console.log('Directories created: 9');
console.log('TypeScript files created: 8');
console.log('Additional files: 1 (.gitkeep)');
console.log('Total items: 18');
console.log('\n✓ All directories and files created successfully!');
